test/lib/flat_mutation_reader_assertions.hh

```diff
@@ -612,8 +643,11 @@ public:
         return *this;
     }
 
-    flat_reader_assertions_v2& produces_row_with_key(const clustering_key& ck) {
+    flat_reader_assertions_v2& produces_row_with_key(const clustering_key& ck, std::optional<api::timestamp_type> active_range_tombstone = std::nullopt) {
```

Comment:
I think taking a tombstone here (defaulted to `tombstone()`) would be more expressive here.

Reply:
it would preclude one from explicitly asserting that there is no active range tombstone, which is actually less expressive

Reply:
`std::optional<tombstone>`, on the other hand...

Reply:
It wouldn't: no range tombstone = current range tombstone has default constructed tombstone.
Note that a stream has an active range tombstone at *any* point. There is an implicit `range_tombstone_change{before_all_clustered_rows, tombstone()}` at the start of every clustering range.

Reply:
I mean you wouldn't be able to distinguish "user asserts there is no active range tombstone" from "user does not care"?

Reply:
Oh, I see. You are right. Indeed, we need `std::optional<tombstone>`, `std::nullopt` meaning "whatever".